alacritty_terminal/src/term/mod.rs

```diff
@@ -372,48 +372,55 @@ impl<T> Term<T> {
             self.mark_fully_damaged();
         }
 
+        // Update tracking of cursor, selection, and vi mode cursor.
+
+        let previous_cursor = mem::replace(&mut self.damage.last_cursor, self.grid.cursor.point);
+        let previous_selection = mem::replace(&mut self.damage.last_selection, selection);
+
+        let display_offset = self.grid().display_offset();
+        let vi_cursor_point = if self.mode.contains(TermMode::VI) {
+            let vi_cursor_point =
+                point_to_viewport(display_offset, self.vi_mode_cursor.point).unwrap();
+            Some(vi_cursor_point)
+        } else {
+            None
+        };
+
+        let previous_vi_cursor_point =
+            mem::replace(&mut self.damage.last_vi_cursor_point, vi_cursor_point);
```

Comment:
```suggestion
        let display_offset = self.grid().display_offset();
        let vi_cursor_point = if self.mode.contains(TermMode::VI) {
            let vi_cursor_point =
                point_to_viewport(display_offset, self.vi_mode_cursor.point).unwrap();
            Some(vi_cursor_point)
        } else {
            None
        };
        
        let previous_cursor = mem::replace(&mut self.damage.last_cursor, self.grid.cursor.point);
        let previous_selection = mem::replace(&mut self.damage.last_selection, selection);
        let previous_vi_cursor_point =
            mem::replace(&mut self.damage.last_vi_cursor_point, vi_cursor_point);
```

I think this way around would be pretty cool, this way the mem replace stuff is all together.